include/caffe/parallel.hpp

```diff
@@ -14,6 +15,12 @@
 #include "caffe/syncedmem.hpp"
 #include "caffe/util/blocking_queue.hpp"
 
+#ifdef USE_NCCL
```

Comment:
I'm a bit confused by the guards here. With the old parallelism removed, shouldn't all of this be guarded?